cmd/pretty_pipe/pretty_pipe.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/rs/zerolog"
+	"os"
+)
+
+func isInputFromPipe() bool {
+	fileInfo, _ := os.Stdin.Stat()
+	return fileInfo.Mode()&os.ModeCharDevice == 0
+}
+
+func main() {
+	if !isInputFromPipe() {
+		fmt.Println("The command is intended to work with pipes.")
+		fmt.Println("Usage: app_with_zerolog |  2> >(pretty_pipe)")
+		os.Exit(1)
+		return
+	}
+
+	scanner := bufio.NewScanner(os.Stdin)
```

Comment:
You don't need a buffer as zerolog guarantees to perform one write() per event.

Reply:
I replaced Scanner with Reader as it has a much smaller buffer (4k vs 64k). I hope this is what you meant - otherwise I would need some more explanation :).

Reply:
Sorry, I meant you don't need a scanner, just a single buffer, each read will get a single event.

Reply:
I think this is ok then. When creating a reader one 4k buffer is allocated and in the loop the content is taken out into a `strings.Builder`.

Reply:
I'm just saying you don't need a scanner to find the \n boundaries. Just pass the read buffer to the console writer. You can even `io.Copy(consoleWriter, io.Stdin)`.

Reply:
Done

Reply:
I tested this tool and I think scanner would still be needed because JSONs might be coming from a file - maybe you stored logs to a file and now you are inspecting the file. So `cat log.jsonl | prettylog` and this does not work currently.